From 318d9ef80a7dc21da7ad45fa46c11c8fb19ec8fb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 19 Dec 2018 16:59:46 +0100
Subject: [PATCH] Filter out too large boxes on selection

Tesseract returns way too large boxes that cover the whole page, mostly containing only a single special character.

This is a complement for commit d1da8cd2 to filter out all of these boxes also on text selection.
---
 paperwork-gtk/src/paperwork/frontend/mainwindow/pages.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/paperwork-gtk/src/paperwork/frontend/mainwindow/pages.py b/paperwork-gtk/src/paperwork/frontend/mainwindow/pages.py
index e324c670..b022b178 100644
--- a/paperwork-gtk/src/paperwork/frontend/mainwindow/pages.py
+++ b/paperwork-gtk/src/paperwork/frontend/mainwindow/pages.py
@@ -731,6 +731,9 @@ class SimplePageDrawer(Drawer):
         index = 0
         for line in self.boxes['lines']:
             for box in line.word_boxes:
+                if box.content.strip() == "" or (box.position[0][0] == 0
+                                                 and box.position[0][1] == 0):
+                    continue
                 rcx = (box.position[0][0] + box.position[1][0]) / 2
                 rcy = (box.position[0][1] + box.position[1][1]) / 2
                 w = box.position[1][0] - box.position[0][0]
@@ -767,6 +770,9 @@ class SimplePageDrawer(Drawer):
         selected = []
         for line in self.boxes['lines']:
             for box in line.word_boxes:
+                if box.content.strip() == "" or (box.position[0][0] == 0
+                                                 and box.position[0][1] == 0):
+                    continue
                 if box == box_start:
                     in_list = True
                 if in_list:
-- 
2.20.0

